app/components/navigation/navigation.tsx

```diff
@@ -1,17 +1,76 @@
 "use client";
-import Link from "next/link";
+import { useState } from "react";
 import { IoHome } from "react-icons/io5";
-import { FaPlusCircle } from "react-icons/fa";
 import OpenAndSave from "../buttons/open-and-save";
-export default function Nav() {
-  return (
-    <div className="flex flex-col items-center h-screen bg-black justify-start gap-4 p-1">
-      <Link href="/">
-        <IoHome size={30} className="fill-white" />
-      </Link>
-      <OpenAndSave>
-        <FaPlusCircle size={30} className="fill-white" />
-      </OpenAndSave>
-    </div>
-  );
+import Link from "next/link";
+import { useLiveQuery } from "dexie-react-hooks";
+import { db } from '../../db/db';
+import {
+	FaPlusCircle,
+	FaCog,
+	FaRegWindowClose,
+} from "react-icons/fa";
+import UniversalSettings from "./settings";
+
+async function addInitialSettings(): Promise<void> {
+	db.settings.count().then(async (count: number) => {
+		if (count === 0) {
+			await db.settings.add({
+				reversed: false,
+				doublePage: false,
+				darkmode: false,
+				displayPerPage: 20,
+			});
+		}
+	});
+}
+
+const Nav: React.FC = () => {
+	addInitialSettings();
+
+	const storedSettings = useLiveQuery(() => db.settings.toArray());
+	const isSettingsLoading = !storedSettings;
+
+	const [showSettings, setShowSettings] = useState<Boolean>(false);
+
+	// Debugging purpose here, to check if the settings is stored correctly.
+	// As addInitialSettings create two data rows at the very first time,
+	// the storedSettings will be an array of two objects.
+	// But, if it's just happened in dev env, it should only check the first object in prod.
+	if (storedSettings) {
+		console.log(storedSettings[0]);
+		// console.log(storedSettings[1]);
+	}
+
+	return (
+		<>
+			<div className="flex flex-col items-center h-screen bg-black justify-start gap-4 p-1">
+				<Link href="/">
+					<IoHome size={30} className="fill-white" />
+				</Link>
+				<OpenAndSave>
+					<FaPlusCircle size={30} className="fill-white" />
+				</OpenAndSave>
+				<FaCog
+					size={30}
+					className="fill-white"
+					onClick={() => setShowSettings(!showSettings)}
+				/>
+			</div>
+			{
+				showSettings && !isSettingsLoading ? (
+					<div className="flex flex-col items-center h-screen bg-black justify-start gap-4 p-1">
+						<FaRegWindowClose
+							size={30}
+							className="fill-white"
+							onClick={() => setShowSettings(!showSettings)}
+						/>
```

Comment:
Can we move close icon to right most side and wrap the icon with button so that mouse cursor effect change ?